--- login.conf	Thu Sep 26 01:07:54 2019
+++ login.conf.bk	Thu Sep 26 01:10:22 2019
@@ -8,7 +8,7 @@
 # Standard authentication styles:
 #
 # passwd	Use only the local password file
-# chpass	Do not authenticate, but change user's password (change
+# chpass	Do not authenticate, but change users password (change
 #		the YP password if the user has one, else change the
 #		local password)
 # lchpass	Do not login; change user's local password instead
@@ -33,7 +33,7 @@
 # The default values
 # To alter the default authentication types change the line:
 #	:tc=auth-defaults:\
-# to read something like: (enables passwd, "myauth", and activ)
+# to be read something like: (enables passwd, "myauth", and activ)
 #	:auth=passwd,myauth,activ:\
 # Any value changed in the daemon class should be reset in default
 # class.
@@ -55,7 +55,7 @@
 #
 # Settings used by /etc/rc and root
 # This must be set properly for daemons started as root by inetd as well.
-# Be sure to reset these values to system defaults in the default class!
+# Be sure reset these values back to system defaults in the default class!
 #
 daemon:\
 	:ignorenologin:\
@@ -70,11 +70,23 @@
 #
 # Staff have fewer restrictions and can login even when nologins are set.
 #
+#staff:\
+	#:datasize-cur=1536M:\
+	#:datasize-max=infinity:\
+	#:maxproc-max=512:\
+	#:maxproc-cur=256:\
+	#:ignorenologin:\
+	#:requirehome@:\
+	#:tc=default:
+
 staff:\
-	:datasize-cur=1536M:\
+	:datasize-cur=4096M:\
 	:datasize-max=infinity:\
-	:maxproc-max=512:\
-	:maxproc-cur=256:\
+	:maxproc-cur=512:\
+	:maxproc-max=1024:\
+	:openfiles-cur=4096:\
+	:openfiles-max=8192:\
+	:stacksize-cur=32M:\
 	:ignorenologin:\
 	:requirehome@:\
 	:tc=default:
@@ -92,9 +104,9 @@
 #
 pbuild:\
 	:datasize-max=infinity:\
-	:datasize-cur=6144M:\
+	:datasize-cur=4096M:\
 	:maxproc-max=1024:\
-	:maxproc-cur=384:\
+	:maxproc-cur=256:\
 	:tc=default:
 
 #
@@ -107,3 +119,7 @@
 unbound:\
 	:openfiles=512:\
 	:tc=daemon:
+
+gnome:\
+        :datasize-cur=1024M:\
+        :tc=default:
