--- /etc/login.conf	Fri Sep 27 09:33:16 2019
+++ /etc/login.conf.new	Fri Sep 27 09:33:42 2019
@@ -40,14 +40,14 @@ auth-ftp-defaults:auth-ftp=passwd:
 #
 default:\
 	:path=/usr/bin /bin /usr/sbin /sbin /usr/X11R6/bin /usr/local/bin /usr/local/sbin:\
-	:umask=022:\
-	:datasize-max=768M:\
-	:datasize-cur=768M:\
-	:maxproc-max=256:\
-	:maxproc-cur=128:\
-	:openfiles-max=1024:\
-	:openfiles-cur=512:\
-	:stacksize-cur=4M:\
+	:umask=077:\
+	:datasize-max=8192M:\
+	:datasize-cur=1024M:\
+	:maxproc-max=1024:\
+	:maxproc-cur=512:\
+	:openfiles-max=8192:\
+	:openfiles-cur=4096:\
+	:stacksize-cur=32M:\
 	:localcipher=blowfish,a:\
 	:tc=auth-defaults:\
 	:tc=auth-ftp-defaults:
